### PR TITLE
run ci and integration test task on arm

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: YangKeao <yangkeao@chunibyo.icu>

### What problem does this PR solve?

Run ci workflow on arm

### CODE REVIEW

File: .github/workflows/integration_test.yml
Comment (WangXiangUSTC) [inline .github/workflows/integration_test.yml:21]: Why use this commit instead of a released version? It might need a comment.

**Review by WangXiangUSTC — COMMENTED** [review]
rest LGTM

File: .github/workflows/ci.yml
Comment (STRRL) [inline .github/workflows/ci.yml:54]: it seems we should also append  `fail-fast: false`

**Comment by YangKeao** [comment]
Blocked by https://github.com/chaos-mesh/chaos-mesh/pull/3258. Though we can use an older version of `kubectl`, using the latest one is more elegant.

**Comment by YangKeao** [comment]
The latest problem is that the kind in dind fail to connect to the network outside.

**Comment by STRRL** [comment]
> The latest problem is that the kind in dind fail to connect to the network outside.

What does it mean? 🙈

**Comment by YangKeao** [comment]
> > The latest problem is that the kind in dind fail to connect to the network outside.
> 
> What does it mean? see_no_evil

We run the action in a dind environment, and start a kind cluster in it. However, the kind cluster seems failed to connect to the internet, and cannot download the image from the dockerhub (or any other mirrors). Only the pod with preloaded image can be booted.

**Comment by YangKeao** [comment]
Some problems found in this PR is also recorded in https://github.com/chaos-mesh/chaos-mesh/issues/3173

File: .github/workflows/integration_test.yml
Comment (YangKeao) [inline .github/workflows/integration_test.yml:36]: The network problem in kind is all about DNS.

`kind` has some magic scripts to fix the DNS loop in coredns when running inside a docker, see https://github.com/kubernetes-sigs/kind/blob/main/images/base/files/usr/local/bin/entrypoint#L375, and so does `minikube`. The awful thing is that: these scripts run, iptables rules are also applied successfully, but the DNS just don't work.

I have entered the environment manually, and dig/nslookup/tcpdump/... the DNS request with a lot of tools, but both the redirection source/target doesn't respond the DNS request. To further understand the problem, we could trace with the iptables, enable debug log of dockerd... It will take a lot more time to find the root, so I choose a much simpler way to solve: replace the dns server to 8.8.8.8 ...

File: test/integration_test/run.sh
Comment (WangXiangUSTC) [inline test/integration_test/run.sh:55]: can remove the space

File: test/integration_test/run.sh
Comment (WangXiangUSTC) [inline test/integration_test/run.sh:45]: OK, I will fix them later

**Review by WangXiangUSTC — APPROVED** [review]
rest LGTM

File: test/integration_test/run.sh
Comment (YangKeao) [reply test/integration_test/run.sh]: I'll also replace the initial tab with spaces (as this file is indented with spaces), to keep align with former and later codes.

**Comment by WangXiangUSTC** [comment]
LGTM

**Comment by cwen0** [comment]
@YangKeao Remember to update the configuration of ti-chi-bot before merging this pr. https://github.com/ti-community-infra/configs/blob/main/prow/config/config.yaml

**Comment by cwen0** [comment]
@YangKeao Remember to update the configuration of ti-chi-bot before merging this pr. https://github.com/ti-community-infra/configs/blob/main/prow/config/config.yaml

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**
